examples/platform/silabs/SiWx917/SiWx917DeviceDataProvider.cpp

```diff
@@ -24,12 +26,183 @@
 
 namespace chip {
 namespace DeviceLayer {
-namespace EFR32 {
+namespace SIWx917 {
+#include <string>
+using namespace std;
 
 // using namespace chip::Credentials;
 using namespace chip::DeviceLayer::Internal;
 
-CHIP_ERROR EFR32DeviceDataProvider::GetSetupDiscriminator(uint16_t & setupDiscriminator)
+#ifdef SIWX917_USE_COMISSIONABLE_DATA
+uint8_t SIWx917DeviceDataProvider::WriteBits(uint8_t * bits, uint8_t offset, uint64_t input, uint8_t numberOfBits,
+                                             uint8_t totalPayloadInBits)
+{
+    if ((offset + numberOfBits) > totalPayloadInBits)
+    {
+        return -1;
+    }
+    uint8_t index = offset;
+    offset += numberOfBits;
+    while (input != 0)
+    {
+        if (input & 1)
+        {
+            bits[int(index / 8)] |= (1 << (index % 8));
+        }
+        index++;
+        input >>= 1;
+    }
+    return offset;
+}
+
+// Generation of the payload using the details provided in the DeviceConfig.h
+void SIWx917DeviceDataProvider::generateQrCodeBitSet(uint8_t * payload)
```

Comment:
Is there a reason this is not using the existing payload generator we have in the tree?

Reply:
Actually this writes in byte by byte format, and I was getting incorrect value by payload generation present in the tree. I was getting the payload value directly (MT:xxxxxxxxxxxxxxxxxxx) and this provides in byte by byte format and flashes that in the NVM section.

Reply:
@chirag-silabs I'm not following the distinction here.  Is the issue that you want to exclude the "MT:" bit?  Or that you want to not do the base-38 encoding?  Or something else?

My instinct is that duplicating the bit-pushing here is very failure-prone and I would really like to understand what prevents us from using the existing code we have that does that and has tests and whatnot.

Reply:
@bzbarsky-apple address the comments on the latest commit